docs: refactor the project documentation (#305)
stellar/java-stellar-anchor-sdk: docs/00 - Stellar Anchor Platform.md

<div>
<img alt="Stellar" src="https://github.com/stellar/.github/raw/master/stellar-logo.png" width="558" />
<br/>
<strong>Creating equitable access to the global financial system</strong>
</div>

# Stellar Anchor Platform

> The Anchor Platform is a web application whose goal is to facilitate Anchor
> integration with the Stellar network in an interoperable way, compiant with the
> [SEP (Stellar Ecosystem Protocols)](https://github.com/stellar/stellar-protocol/tree/master/ecosystem).
> 
> The Anchor Platform will be exposing public endpoints and APIs that provide
> functionalities described in the [SEPs] while abstracting most of the business
> logic necessarry to enable interoperability with the Stellar network.
> 
> This way, Anchors willing to integrate with Stellar can use the Anchor Platform
> to speed up their development process, allowing them to focus mainly on the
> business logic that's specific to their businesses and use cases.

Reviewer: I think content here is good, but I think we can be a bit stronger with the language used to imply that this is the best way to run an anchor.

```suggestion
The Anchor Platform is the easiest and fastest way to deploy a [SEP-compatible](https://github.com/stellar/stellar-protocol/tree/master/ecosystem) anchor service. 

It implements a the majority of standardized API (SEP) endpoints that wallets, exchanges, and other applications use, and provides a set of backend HTTPS APIs & callbacks for the anchor to integrate with for specifying fees, exchange rates, and off-chain transaction status updates.

The goal of the Anchor Platform is to abstract all Stellar-specific functionality and requirements for running an anchor, allowing businesses to focus on the core business logic necessary to provide these services.
```

## Table of Content

The full documentation can be found under the [`docs` directory](/docs), under the structure:

- [00 - Stellar Anchor Platform](/docs/00%20-%20Stellar%20Anchor%20Platform.md)
- [01 - Running & Configuring the Application](/docs/01%20-%20Running%20%26%20Configuring%20the%20Application)
  - [A - Running & Configuring the Application](/docs/01%20-%20Running%20%26%20Configuring%20the%20Application/A%20-%20Running%20%26%20Configuring%20the%20Application.md)
  - [B - Circle Payment Observer](/docs/01%20-%20Running%20%26%20Configuring%20the%20Application/B%20-%20Circle%20Payment%20Observer.md)
- [02 - Contributing](/docs/02%20-%20Contributing)
  - [A - CONTRIBUTING.md](/docs/02%20-%20Contributing/A%20-%20CONTRIBUTING.md)
  - [B - Developer Tools](/docs/02%20-%20Contributing/B%20-%20Developer%20Tools.md)
  - [C - Logging Guidelines](/docs/02%20-%20Contributing/C%20-%20Logging%20Guidelines.md)
  - [D - Database Migration](/docs/02%20-%20Contributing/D%20-%20Database%20Migration.md)
  - [E - Publishing the SDK](/docs/02%20-%20Contributing/E%20-%20Publishing%20the%20SDK.md)

> - [03 - Anchor Integration](/docs/03%20-%20Anchor%20Integration) // TODO

Reviewer: Maybe "Implementing the Anchor Server" since thats what we refer to it as?

- [04 - Subprojects Usage](/docs/04%20-%20Subprojects%20Usage) // TODO

## Glossary

Here are the important terminology used in this project:

- **Anchor**: on/off ramps of the Stellar network. More information is available [here](https://developers.stellar.org/docs/anchoring-assets/).
- **Wallet**: a frontend application used to interact with the Stellar network on behalf of a user.
- **Sending Anchor**: a therminology used in the context of [SEP-31]. Refers to an entity that receives funds from a user and forwards it (after taking a fee) to a receiving anchor, in the SEP-31 `Sending Client->Sending Anchor->Receiving Anchor-> Receiving Client` flow.

> - **Receiving Anchor**: a therminology used in the context of [SEP-31]. Refers to an entity that receives funds from a user and forwards it (after taking a fee) to a receiving anchor, in the SEP-31 `Sending Client->Sending Anchor->Receiving Anchor-> Receiving Client` flow. It's the entity the Platform SEP-31 service is currently built for.

Reviewer: ```suggestion
- **Receiving Anchor**: a terminology used in the context of [SEP-31]. Refers to an entity that receives funds from a user and forwards it (after taking a fee) to a receiving client (or recipient), in the SEP-31 `Sending Client->Sending Anchor->Receiving Anchor-> Receiving Client` flow. This is what the Anchor Platform currently implements.
```

> - **Ecosystem Players**: any entity or end-user that's part of the Stellar ecosystem. This includes Anchors, Wallets, Users, Issuers, Payment providers, and others.

Reviewer: We don't commonly use this phrase, do we need it in the glossary? Maybe we just define "Ecosystem"?

> - **Anchor Platform (or Platform)**: the web application that will be exposing public endpoints and APIs. It will be compliant with the [SEPs] to guarantee interoperability in the Stellar network and will delegate business-specific logic to the Anchor Server.

Reviewer: grammar nit: can we change the phrasing "will be" (future-tense) to "is" (present-tense)? This happens a few times in the doc.

```suggestion
- **Anchor Platform (or Platform)**: the web application that will be exposing public endpoints and APIs. It is compliant with the [SEPs] to guarantee interoperability in the Stellar network and delegates business-specific logic to the Anchor Server.
```

> - **Anchor Server**: a microservice that will be responsible for the Anchor-specific business logic used in the the Anchor Platform. This service will be reached out by the Anchor Platform to perform some actions like:

Reviewer: ```suggestion
- **Anchor Server**: a microservice that will be responsible for the Anchor-specific business logic used in the the Anchor Platform. This service interacts with the Anchor Platform to perform some actions like:
```

- **Platform API (`Sync Anchor->Platform`)**: a syncronous API that the Anchor can use to fetch information (e.g. transactions or quotes) and also update transactions stored in the Platform database.

## Microservices

In order to deploy this project, you'll need to have the following microservices running:

- **Database Server**: usually, you'll use a relational database like MySQL or PostgreSQL, but we also support SQLite, commonly used in local development.
- **Queue Service**: we currently support [Kafka](https://kafka.apache.org/) and [Amazon SQS](https://aws.amazon.com/sqs/).
- **Anchor Platform Server**: this is the main application that will be providing public endpoints for your Anchor application.
- **Anchor Server**: this is the microservice that will be responsible for the Anchor-specific business logic used in the the Anchor Platform.

## Architecture

The following image shows the architecture of the Anchor Platform, as well as how it interacts with the Anchor Server and the Wallet/Client/Sending Anchor.

> ![img.jpeg](/docs/resources/img/anchor-platform-components-architecture.jpeg)

Reviewer: This image is gold.

Author: Praise @reecexlm for it 😄

> As you can see, the Anchor Platform receives interactions from ecosystem players and deals with the interoperability part described in the SEPs. The Anchor Server is only called when there is a pending action to be performed.

Reviewer: Ah, I see this is where we say "ecosystem players". Can we just do "ecosystem participants"?
